### PR TITLE
Made updates to documentation

### PR DESCRIPTION
**Pull request by Nolan-Steed**
@Melkiades I tried adding the "@seealso" to the documentation to have links to other packages that are being used but I was having issues implementing it. I might need your help with that. 

@gmbecker The "Horizontal Pagination" example in `paginate_listings.R` doesn't seem to work on my end no matter what value I input for it, I get this error. 
image

### CODE REVIEW

**Comment by Melkiades** [comment]
if you do `verbose = TRUE`, you get why there is an errror:
```
> paginate_listing(lsting, cpp = 10, lpp = 40, verbose = TRUE)
Checking pagination after row 25
	....................... OK [25 lines]
Adjusted characters per page: 10 [original: 10, table inset: 0]
Checking pagination after column 6
	....................... FAIL: columns take up too much space (134 characters)
Checking pagination after column 5
	....................... FAIL: columns take up too much space (119 characters)
Checking pagination after column 4
	....................... FAIL: columns take up too much space (89 characters)
Checking pagination after column 3
	....................... FAIL: columns take up too much space (63 characters)
Checking pagination after column 2
	....................... FAIL: columns take up too much space (34 characters)
Checking pagination after column 1
	....................... FAIL: columns take up too much space (28 characters)
Error in find_pag(pagdf, start, guess, rlpp = adjrlpp, min_siblings = min_siblings, :
Unable to find any valid pagination between 1 and 6
```

File: R/paginate_listing.R [hunk 1]
```diff
@@ -25,13 +25,7 @@
 #' ) %>%
 #'   add_listing_col("AETOXGR") %>%
 #'   add_listing_col("BMRKR1", format = "xx.x") %>%
-#'   add_listing_col("AESER / AREL", fun = function(df) paste(df$AESER, df$AREL, sep = " / "))
-#'
-#' # Vertical pagination
-#' paginate_listing(lsting, lpp = 10)
-#'
-#' # Horizontal pagination
-#' # paginate_listing(lsting, cpp = 10, lpp = 40)
```
Comment (Melkiades) [inline R/paginate_listing.R:34]: add a paginate_listing with cpp and lpp

File: R/rlistings.R [hunk 2]
```diff
@@ -13,25 +13,56 @@ setOldClass(c("MatrixPrintForm", "list"))
...
-#' lsting <- as_listing(dat[1:25, ], key_cols = c("USUBJID", "AESOC")) %>%
+#' lsting <- as_listing(dat[1:25, ],
+#'   key_cols = c("USUBJID", "AESOC"),
+#'   main_title = "Example Title for Listing",
+#'   subtitles = "This is the subtitle for this Adverse Events Table",
+#'   main_footer = "Main footer for the listing",
+#'   prov_footer = c(
+#'     "You can even add a subfooter", "Second element is place on a new line",
+#'     "Third string"
+#'   )
```
Comment (Melkiades) [inline R/rlistings.R:44]: I would add these parameters only one time to show how they work and not in multiple examples (as they are long and not giving more info if repeated)

File: R/rlistings.R [hunk 3]
```diff
@@ -89,28 +121,31 @@ as_keycol <- function(vec) {
...
 #' @rdname listings
+#' @export
 get_keycols <- function(df) {
   names(which(sapply(df, is_keycol)))
 }
 
-#' @export
+
 #' @inheritParams formatters::matrix_form
+#' @seealso [formatters::matrix_form()]
```
Comment (Melkiades) [inline R/rlistings.R:144]: Good! You may also add a short description specifying that this is partially inherited from formatters' function

**Review by Melkiades — CHANGES_REQUESTED** [review]
Looks promising to me. I would just tweak a couple of things (see comments) and it is good to go imo. 

ps: try to have very direct examples, minimizing repetitions

File: R/rlistings.R [hunk 4]
```diff
@@ -13,22 +13,51 @@ setOldClass(c("MatrixPrintForm", "list"))
...
 #' @param prov_footer character or NULL. A vector of provenance strings
-#'   for the listing, or `NULL` (the default).
-#'
+#'   for the listing, or `NULL` (the default). Each string element is placed on a new line.
 #' @return A `listing_df` object, sorted by the key columns.
 #'
+#' @details
+#' key_cols is useful when a patient experiences more than 1 adverse event, rather than
+#' listing out a separate row for each adverse event, they are able to be grouped
+#' together for each patient. These columns are always present in the listing even if
```
Comment (Melkiades) [inline R/rlistings.R:30]: even if? I would keep all of this a bit more general, as it is not always patients or AE we want to have as key_cols

**Review by Melkiades — CHANGES_REQUESTED** [review]
Lgtm! well done. Just fix the one comment I added and it is done for me :)

**Review by Melkiades — APPROVED** [review]
Lgtm! thanks Nolan. @gmbecker, can you take a look at this PR? thanks in advance ^^ ofc to merge after refactoring @Nolan-Steed and relative updates

**Comment by gmbecker** [comment]
Hi @Nolan-Steed @Melkiades,

So this is likely to conflict with #69 because I did significantly expand the core documentation there ( since I had to change the name of the documented arguments anyway), we will have to wait for that to land and then look at the overlap, but that should land today, so we'll know soon

**Review by shajoezhu — CHANGES_REQUESTED** [review]
hi @Nolan-Steed , we have a few updates from the main. can you please help to update your code, and resolve these conflicts.

**Review by Melkiades — CHANGES_REQUESTED** [review]
@Nolan-Steed can I ask you to add examples for the new parameters? e.g. `disp_cols`... That would be great!! thanks :)

**Review by shajoezhu — APPROVED** [review]
lgtm! thanks @Nolan-Steed

**Review by Melkiades — APPROVED** [review]
Also for me, it seems all fine. @gmbecker can you take a look?

File: NAMESPACE [hunk 5]
```diff
@@ -6,10 +6,13 @@ export(add_listing_col)
 export(add_listing_dispcol)
 export(as_keycol)
 export(as_listing)
+export(basic_run_lens)
```
Comment (Melkiades) [inline NAMESPACE:9]: the new exported functions should remain internals

File: R/rlistings.R [hunk 6]
```diff
@@ -303,6 +331,7 @@ add_listing_col <- function(df, name, fun = NULL, format = NULL, na_str = "-") {
     }
 
 
+
```
Comment (gmbecker) [inline R/rlistings.R:334]: please avoid unnecessary whitespace changes

File: R/rlistings_methods.R [hunk 7]
```diff
@@ -15,17 +15,21 @@
...
 print.listing_df <- function(x, ...) {
   cat(toString(matrix_form(x)))
   invisible(x)
 }
 
 
 ## because rle in base base is too much of a stickler for being atomic
+#' @rdname listing_methods
+#' @param x listing_df. The listing.
+#' @export
```
Comment (gmbecker) [inline R/rlistings_methods.R:32]: why are we exporting this?

File: R/rlistings_methods.R [hunk 8]
```diff
@@ -200,18 +215,21 @@ setMethod(
...
 
 #' @rdname listing_methods
+#' @param obj The object.
 #' @export
 setMethod(
   "subtitles", "listing_df",
   function(obj) attr(obj, "subtitles") %||% character()
 )
 #' @rdname listing_methods
+#' @param obj The object.
```
Comment (gmbecker) [inline R/rlistings_methods.R:225]: This does not need to be repeated every time when al the methoeds are sharing an rd file

File: R/rlistings_methods.R [hunk 9]
```diff
@@ -239,6 +257,7 @@ setMethod(
 
 #' @rdname listing_methods
 #' @param obj The object.
+#' @param value Character string.
```
Comment (gmbecker) [inline R/rlistings_methods.R:260]: Also should not be repeated

File: R/paginate_listing.R [hunk 10]
```diff
@@ -19,20 +19,20 @@
...
+#' mat <- matrix_form(lsting)
+#'
+#' cat(toString(mat))
 #'
 #' # Vertical pagination
 #' paginate_listing(lsting, lpp = 10)
 #'
-#' # Horizontal pagination
-#' # paginate_listing(lsting, cpp = 10, lpp = 40)
+#' #Horizontal pagination
```
Comment (gmbecker) [inline R/paginate_listing.R:31]: That second `#` shouldn't be there.

File: R/rlistings_methods.R [hunk 11]
```diff
@@ -37,7 +41,11 @@ basic_run_lens <- function(x) {
   diff(c(0L, i))
 }
 
-
+#' @rdname listing_methods
+#' @param df listing_df. The listing.
+#' @param colnm Column name
+#' @param colvec Column values based on colnm
+#' @export
```
Comment (gmbecker) [inline R/rlistings_methods.R:48]: I don't think there's much value in exporting this.

File: R/rlistings_methods.R [hunk 12]
```diff
@@ -49,8 +57,14 @@ format_colvector <- function(df, colnm, colvec = df[[colnm]]) {
     strvec
 }
 
+#' @rdname listing_methods
+#' @param vec A vector.
+#' @export
```
Comment (gmbecker) [inline R/rlistings_methods.R:62]: If this were going to be exported I would expect it to also be moved to `formatters` (which it may eventually need to be anyway)

File: R/rlistings_methods.R [hunk 13]
```diff
@@ -69,9 +83,10 @@ setMethod("vec_nlines", "ANY", function(vec, max_width = NULL) {
 ##   ret[is.na(ret)] <- format_value(NA_character
 ## })
 
+
```
Comment (gmbecker) [inline R/rlistings_methods.R:86]: whitespace

**Review by gmbecker — CHANGES_REQUESTED** [review]

File: R/rlistings_methods.R (hunk 12 above)
Comment (Nolan-Steed) [reply R/rlistings_methods.R:62]: Should I just remove the export on line 62? Or is the best practice to remove all of the roxygen lines?

Comment (gmbecker) [reply R/rlistings_methods.R:57]: just the export is fine, though in this case the other lines aren't doing very much in practice...

File: R/rlistings_methods.R [hunk 14]
```diff
@@ -192,7 +198,6 @@ setMethod(
 )
 
 #' @rdname listing_methods
-#' @param obj The object.
```
Comment (Melkiades) [inline R/rlistings_methods.R:195]: This was the culprit that caused the error message @Nolan-Steed

Comment (Nolan-Steed) [reply R/rlistings_methods.R]: Thanks @Melkiades for your help with this!

**Comment by shajoezhu** [comment]
hey guys, can we merge in this PR nowish, and start a new issue and PR to address the remaining problems please. many thanks! @Nolan-Steed @Melkiades @gmbecker

File: NAMESPACE [hunk 15]
```diff
@@ -10,6 +10,7 @@ export(get_keycols)
 export(is_keycol)
 export(listing_dispcols)
 export(paginate_listing)
+export(vec_nlines)
```
Comment (gmbecker) [inline NAMESPACE:13]: Why are we exporting this now?

**Review by gmbecker — CHANGES_REQUESTED** [review]

**Comment by gmbecker** [comment]
> hey guys, can we merge in this PR nowish, and start a new issue and PR to address the remaining problems please. many thanks! @Nolan-Steed @Melkiades @gmbecker

One of the issues is that it is exporting something that wasn't exported before. That is not something we can undo later, because once its exported in a release it is officially part of the API, so I'd say we can't merge it in as it exists now @shajoezhu Other than that, a cursory look at it suggests that it seems ok.

**Review by Melkiades — APPROVED** [review]
@gmbecker Now there are no differences in API (export/internal)

**Comment by Melkiades** [comment]
@gmbecker @cicdguy I do not know why the styler is automatically fixing this and formatters' PR atm ([here](https://github.com/insightsengineering/formatters/pull/88#discussion_r1105580984)). Having partial changes of style might lead to a non-uniform style. I am also fine with having `rlistings` and `formatters` restyled with tidyverse style or adding the current style to the checks. What do you think Gabe?

**Comment by gmbecker** [comment]
> > @gmbecker @cicdguy I do not know why the styler is automatically fixing this and formatters' PR atm ([here](https://github.com/insightsengineering/formatters/pull/88#discussion_r1105580984)). Having partial changes of style might lead to a non-uniform style. I am also fine with having `rlistings` and `formatters` restyled with tidyverse style or adding the current style to the checks. What do you think Gabe?
> 
> We can turn off auto-styling by setting this to `false`: https://github.com/insightsengineering/rlistings/blob/main/.github/workflows/check.yaml#L70-L71
> 
> Totally up to you guys to decide on whether you want it enabled or not. Ideally it'd be nice to follow a standard styling convention across the board.

We discussed this in our standup and the take away was to disable it for now, because piecemeal styling is worse internally-consistent styling, and if/when we have an issue to restyle the whole package a component of that issue will be to turn it back on @cicdguy

**Comment by Melkiades** [comment]
@cicdguy, I do not know if I removed the right auto-update but it complains still:
```
Run cat(paste(
☠ One or more files had styling errors. Please see the log above for remediations, or simply run the following commands for an immediate fix:
────────────────────────────────────────
styler::style_file(c("R/paginate_listing.R", "R/rlistings.R", "R/rlistings_methods.R", "tests/testthat/test-listings.R"))
────────────────────────────────────────
Error: Process completed with exit code 1.
```

**Comment by Melkiades** [comment]
> Ok the auto-update is gone, but the styler workflow still exists to it runs a check and asks the user to update the files manually. In this case you can simply remove the styler workflow completely.

For general reference, I created an issue to get the automated styler back: https://github.com/insightsengineering/sme-tasks/issues/593

**Comment by Melkiades** [comment]
Oh no ahah why is this happening?? can we go back to 6dedebd61fc6c580602fd2360e66d38a4fbcb4dd

**Comment by Melkiades** [comment]
I am going to create a new branch because here we created a mess :D

**Comment by shajoezhu** [comment]
hi @Melkiades , let's rebase this branch, squash it, raise a new PR

**Comment by shajoezhu** [comment]
in fact, we can restyle rlistings repo now. as it is planned to make this package production ready by the end of this increment. we are not in feature freeze right now. we can go ahead with necessary changes. we can deal with restyling rtables and formatters seperately in the next increment
